SIL.Core/Linq/MiscOperators.cs

```diff
@@ -42,5 +42,29 @@ public static void ForEach<TSource> (this IEnumerable<TSource> source, Action<TS
 			foreach (TSource element in source)
 				action (element);
 		}
+
```

Comment:
GlyssenEngine.Utilities.LinqExtensions has a similar method called OnlyOrDefault, except my version of it does not take a predicate. However, some of the places I use it could benefit from the predicate version. (I just use a Where to pre-process the enumeration.) I think we should have a consistent name and put them in the same place (here). I think I have a slight preference for my name since it is more similar to SingleOrDefault and is clear about what happens when there is more than one. If you agree, I'll do my part o move mine over. I'm not 100% sure if it makes sense to keep the implementations distinct or if it would be better to have a single underlying implementation.